pennylane/qaoa/cost.py

```diff
@@ -257,6 +257,8 @@ def maxcut(graph):
         [-0.5 for e in graph.edges], [qml.Identity(e[0]) @ qml.Identity(e[1]) for e in graph.edges]
     )
     H = edge_driver(graph, ["10", "01"]) + identity_h
+    # store the valuable information that all observables are in one commuting group
+    H.grouping_indices = [list(range(len(H.ops)))]
```

Comment:
Hypothetical, but does future transformations (such as simplify) destroy the grouping information?

Reply:
Yes! I think we do not get around this, it's too dangerous.

Reply:
Makes sense!